src/db.rs

```diff
@@ -537,6 +561,94 @@ mod tests {
         kv_store.close().await.unwrap();
     }
 
+    #[tokio::test]
+    async fn test_get_with_object_store_cache() {
+        let object_store: Arc<dyn ObjectStore> = Arc::new(InMemory::new());
+        let mut opts = test_db_options(0, 1024, None);
+        let temp_dir = tempfile::Builder::new()
+            .prefix("objstore_cache_test_")
+            .tempdir_in("/tmp")
+            .unwrap();
+        let db_stats = Arc::new(DbStats::new());
+        let part_size = 1024;
+        let cached_object_store = Arc::new(CachedObjectStore::new(
+            object_store.clone(),
+            temp_dir.path().to_path_buf(),
+            part_size,
+            db_stats.clone(),
+        ));
+
+        opts.object_store_cache_options.root_folder = Some(temp_dir.into_path());
+        let kv_store = Db::open_with_opts(
+            Path::from("/tmp/test_kv_store_with_cache"),
+            opts,
+            cached_object_store.clone(),
+        )
+        .await
+        .unwrap();
+
+        let access_count0 = kv_store.metrics().object_store_cache_part_access.get();
+        let key = b"test_key";
+        let value = b"test_value";
+        kv_store.put(key, value).await;
+        kv_store.flush().await.unwrap();
+
+        let got = kv_store.get(key).await.unwrap();
+        let access_count1 = kv_store.metrics().object_store_cache_part_access.get();
+        assert_eq!(got, Some(Bytes::from_static(value)));
+        assert!(access_count1 > 0);
+        assert!(access_count1 >= access_count0);
+        assert!(kv_store.metrics().object_store_cache_part_hits.get() >= 1);
+
+        assert_eq!(
+            cached_object_store
+                .list(None)
+                .collect::<Vec<_>>()
+                .await
+                .into_iter()
+                .filter_map(Result::ok)
+                .map(|meta| meta.location.to_string())
+                .collect::<Vec<_>>(),
+            vec![
+                "tmp/test_kv_store_with_cache/manifest/00000000000000000001.manifest".to_string(),
+                "tmp/test_kv_store_with_cache/manifest/00000000000000000002.manifest".to_string(),
+                "tmp/test_kv_store_with_cache/wal/00000000000000000001.sst".to_string(),
+                "tmp/test_kv_store_with_cache/wal/00000000000000000002.sst".to_string(),
+            ],
+        );
+
+        // check the files are cached as expected
+        let tests = vec![
+            (
+                "tmp/test_kv_store_with_cache/manifest/00000000000000000001.manifest",
+                0,
+            ),
+            (
+                "tmp/test_kv_store_with_cache/manifest/00000000000000000002.manifest",
+                2,
+            ),
+            (
+                "tmp/test_kv_store_with_cache/wal/00000000000000000001.sst",
```

Comment:
i guess the reason here is that the cache works on `GET` and not covered `PUT` yet, and the GET happens when WAL is being replayed.

i tried add a `kv_store.get()` but failed to cover the cache hit, as it seems that WAL is kept in memory and never hit back. it might worths find a more sophisticated way to cover the cached KV in test case...